test/service.go

```diff
@@ -65,9 +66,52 @@ func CreateService(ctx *Context, name, namespace, image string) (*servingv1beta1
 	return service, nil
 }
 
+func WaitForControllerEnvironment(ctx *Context, ns, envName string) error {
+	return wait.PollImmediate(Interval, 10*time.Minute, func() (bool, error) {
+		pods, _ := ctx.Clients.Kube.CoreV1().Pods(ns).List(metav1.ListOptions{
+			LabelSelector: "app=controller",
+		})
+		for i := range pods.Items {
+			for _, container := range pods.Items[i].Spec.Containers {
+				for _, e := range container.Env {
+					if e.Name == envName && e.Value != "" {
```

Comment:
Can we check for the actual value here? I think we want to check that this is exactly the configured proxy when configuring it and that this is unset when taking it out again.